pkg/config/auth.go

```diff
@@ -0,0 +1,115 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// Client returns a OAuth2 HTTP client based on the configuration for a tenant.
+func (t *TenantConfig) Client(ctx context.Context, logger log.Logger) (*http.Client, error) {
```

Comment:
I think this would fit better in `config.go` where we have all other methods defined on this struct.